debian/mariadb-server-10.5.postinst

```diff
@@ -103,6 +103,14 @@ EOF
 	    rmdir $mysql_upgradedir 2>/dev/null || true
 
     done
+    # Upgrading from mysql.com needs might have the root user as auth_socket.
+    # As this is needed to perform mariadb-upgrade, and mariadb uses unix_socket,
+    # change it now (MDEV-22678).  To keep the impact minimal, we skip innodb
+    # and set key-buffer-size to 0 as it isn't used.
+    if [ -f "$mysql_datadir"/auto.cnf ]; then
```

Comment:
Do we need to check for  `auto.cnf` to detect MySQL 5.7? It is a conditional update, can't we just run it always? And it is cheap to run, right? If `mariadbd` is expensive to start, then the solution should maybe be something that runs as part of `mariadbd` itself by default?